### PR TITLE
Nc news 8

### PR DESCRIPTION
**Pull request by XingWei0920**
I finished task 8

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -222,3 +222,41 @@ describe("/api/articles/:article_id/comments", () => {
    });
   });
 
+  describe("/api/articles/:article_id", () => {
+    test("PATCH:200 update an specific article's vote property", () => {
+      const newVote={inc_vote:20}
```
Comment (popatre) [inline __tests__/app.test.js]: ticket asks for this to be `inc_votes` - with an s on the end.

File: __tests__/app.test.js [hunk 2]
```diff
@@ -222,3 +222,41 @@ describe("/api/articles/:article_id/comments", () => {
...
+     test("GET:404 sends an Not Found error if the article_id does not exists", () => {
+      return request(app)
+        .patch("/api/articles/99")
+        .expect(404)
+        .then((response) => {
+          expect(response.body.msg).toBe("Not Found");
+        });
+     });
+  })  
+
```
Comment (popatre) [inline __tests__/app.test.js:273]: Missing test: Status 400: Bad request, incomplete body
E.g. inc_votes value is not a number.

File: controllers/server-controllers.js [hunk 3]
```diff
@@ -59,6 +59,22 @@ exports.getArticleById= (req, res, next) => {
...
 }
 
+exports.patchArticleById= (req, res, next) => {
+    const newVote=req.body
+    const {article_id}=req.params
+    const commentPromise=[updateArticleById(newVote,article_id)]
+    if (article_id)
+    {
+        commentPromise.push(checkArticle_idExists(article_id))
+    }
```
Comment (popatre) [inline controllers/server-controllers.js]: Do you need this if statement? Will there even not be an article id, given that the route is `/api/articles/:article_id`??

File: models/server-models.js [hunk 4]
```diff
@@ -50,3 +50,11 @@ exports.selectArticleById= (article_id) => {
       });
 }
 
+exports.updateArticleById= (newVote,article_id) => {
+    const {inc_vote}=newVote
```
Comment (popatre) [inline models/server-models.js]: `inc_votes`

**Review by popatre — CHANGES_REQUESTED** [review]
Missing a test and i've left a comment about some code not needed

- [ ] Status 400: Bad request, incomplete body - E.g. inc_votes value is not a number.

**Review by nc-mick-fay — APPROVED** [review]
Good changes, merge into main when you're ready!

Your test name for your new test says it is 404, but you're testing for 400. Change that before you merge

**Pull request merged**